### PR TITLE
refactor: prepare for multi-return values in GenerateForAnnotation

### PR DESCRIPTION
**Pull request by kevmoo**
JsonSerializableGenerator now returns String instead of Future<String>
GeneratorHelper is now private and in the same file is the Generator
Encoding and decoding have been split into mixin classes
  These two aspects of the generator are nicely isolated hopefully
  making future fixes and additions easier

Many members of utils.dart have been moved closer to where they are used
  and made private when possible

Updated tests to be sync now that the core tested function is sync

### CODE REVIEW

File: json_serializable/lib/src/decode_helper.dart [hunk 1]
```diff
@@ -0,0 +1,302 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:source_gen/source_gen.dart';
+
+import 'helper_core.dart';
+import 'json_literal_generator.dart';
+import 'type_helper.dart';
+import 'utils.dart';
+
+abstract class DencodeHelper implements HelperCore {
```
Comment (jakemac53) [inline json_serializable/lib/src/decode_helper.dart]: nit: `DecodeHelper`

File: json_serializable/lib/src/decode_helper.dart [hunk 2]
```diff
@@ -0,0 +1,302 @@
+import 'package:analyzer/dart/element/element.dart';
```
Comment (jakemac53) [inline json_serializable/lib/src/decode_helper.dart]: copyright

**Comment by kevmoo** [comment]
> Did not re-review internals, this is mostly just moves right? 

Yup. Mostly copy-paste. The mixin thing is pretty nice...

**Comment by jakemac53** [comment]
Ya, I like the new structure more for sure

**Pull request merged**